util/io/paths.go

```diff
@@ -58,3 +59,14 @@ func (p *RandomizedTempPaths) GetPathIfExists(key string) string {
 	}
 	return ""
 }
+
+// GetPaths gets a copy of the map of paths.
+func (p *RandomizedTempPaths) GetPaths() map[string]string {
+	p.lock.Lock()
```

Comment:
Is locking really necessary in this get operation?

Reply:
Added a test which fails on go's race detection without this lock.

I switched to a RWLock, since that makes more sense here.